addons/mail/static/src/models/partner.js

```diff
@@ -343,7 +343,10 @@ registerModel({
          * @returns {string|undefined}
          */
         _computeDisplayName() {
-            return this.display_name || this.user && this.user.display_name;
+            if (typeof this.display_name !== 'undefined') {
```

Comment:
we never use typeof to check for undefined

Reply:
also if display_name contains any other falsy value, it should also fallback on the rest